libs/icons/scripts/figma_icons/figma-icon-export.ts

```diff
@@ -0,0 +1,740 @@
+ * @param filePath - relattive path to the file
+ * @returns - SvgDiffResult
+ */
+const buildBeforeAndAfterSvg = async (status: string, filePath: string):  Promise<SvgDiffResult> => {
+  const filename = path.basename(filePath);
+  let beforeSvg = null;
+  let afterSvg = null;
+
+  switch(status) {
+    case 'D':
+      beforeSvg = await getFileContentsFromGit(filePath);
+      break;
+
+      case 'M':
+      beforeSvg = await getFileContentsFromGit(filePath);
+      afterSvg = await getFileContentsFromDisk(filename)
+      break;
+
+    case '':
+      afterSvg = await getFileContentsFromDisk(filename);
+      break;
+  }
+
+
+  return {
+    filename,
+    status,
+    before: beforeSvg,
+    after: afterSvg,
+  }
+}
+
+/**
+ * Generates a Header and table
+ *
+ * @param sectionName - name used for the Header
+ * @param data  - an array of {@link SVGDiffResult}
+ * @returns The string of html used to represent a section e.g Added in the Changelog
+ */
+const buildHTMLSection = (sectionName: string, data: Array<SvgDiffResult>) => {
+  const content = `<h2>${sectionName}</h2>`
+  const table: string = buildHTMLTable(data);
+
+  return [content, table].join('\n')
+}
+
+/**
+ * Generates a HTML Table
+ *
+ * @param data  an array of {@link SvgDiffResult}
+ * @returns - The html table markup
+ */
+const buildHTMLTable = (data: Array<SvgDiffResult>) => {
+  const tableRows = data.map((diff) => `<tr>
+  <td>${diff.filename}</td>
+  <td>${diff.before || ''}</td>
+  <td>${diff.after || ''}</td>
+</tr>`);
+
+  const tableBody = `<tbody>
+    ${tableRows.join('')}
+  </tbody>`
+
+  const table = `<table>
+  <thead>
+    <tr>
+      <td>Filename</td>
+      <td>Before</td>
+      <td>After</td>
+    </tr>
+  </thead>
+  ${tableBody}
+</table>`
+
+  return table;
+}
+
+
+/**
+ * Creates the axios client with the appropriate
+ * headers and url.
+ *
+ * @param apiToken required token to pass in the headers
+ * @returns Axios client object
+ */
+const client = (apiToken) => {
+  const figmaBaseApiURL = 'https://api.figma.com/v1';
+  const instance = axios.create({
+    baseURL: figmaBaseApiURL,
+    headers: {
+      'Content-Type': 'application/json',
+      'X-Figma-Token': apiToken
+    }
+  })
+
+  instance.interceptors.request.use((conf) => {
+    conf.startTime = new Date().getTime();
+
+    return conf;
+  });
+
+  return instance;
+};
+
+/**
+ * Commits the files to a newly created branch
+ * @param files - a list of files from GitStatus
+ */
+const commitChanges = async (files) => {
+  const git = gitClient();
+  const branchName = `nightly/${strDate}-icon-update`;
+
+  await git.checkoutLocalBranch(branchName);
+
+  await git.add(files.map(file => `${path.basename(file.path)}`));
+  await gitClient({ baseDir: srcDir}).add(['icon-data.json']);
+  await gitClient().add(`${baseDir}/changelogs`);
+  await git.commit(['ci(nightly): figma icons update', `number of icons updated: ${files.length}`]);
```

Comment:
Not a blocker, potential follow-up: thoughts on appending the same date - version format used in the changelog template to this commit message? It may help to distinguish icon updates as they stack up, especially if update requests continue to be ad hoc rather than batched/scheduled.